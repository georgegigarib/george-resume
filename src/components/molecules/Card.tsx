import { CardBody } from '@/components/atoms/CardBody'
import { CardContainer } from '@/components/atoms/CardContainer'

interface Props {
  content: React.ReactNode
}

export default function Card({ content = '' }: Props) {
  return (
    <CardContainer className="inter-var">
      <CardBody className="bg-gray-50 group/card  dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl  border">
        {content}
      </CardBody>
    </CardContainer>
  )
}
