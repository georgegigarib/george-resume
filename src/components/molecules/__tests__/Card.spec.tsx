import { render } from '@testing-library/react'
import { describe, expect, it } from 'vitest'
import Card from '@/components/molecules/Card'

describe('Card component', () => {
  it('renders the content passed as props', () => {
    const { getByText } = render(<Card content="Test Content" />)
    const contentElement = getByText('Test Content')

    expect(contentElement).toBeTruthy()
  })

  it('renders with the correct classes', () => {
    const { container } = render(<Card content="Test Content" />)

    expect(container.querySelector('.bg-gray-50')).toBeTruthy()
  })
})
