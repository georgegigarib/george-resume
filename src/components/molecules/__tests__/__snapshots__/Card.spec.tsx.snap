// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex w-full h-full"
    style="perspective: 1000px;"
  >
    <div
      class="w-full relative transition-all duration-200 ease-linear inter-var"
      style="transform-style: preserve-3d;"
    >
      <div
        class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
      >
        Test Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
